### PR TITLE
feat: Introduce go-version file to support multiple go versions on CI

### PR DESCRIPTION
**Pull request by toninis**
#### Summary

Introduce `.go-version` with the minor version of golang we are using to be consistent across the releases we support.

#### Related PR
https://github.com/mattermost/delivery-platform/pull/44

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by isacikgoz — APPROVED** [review]

File: .go-version [hunk 1]
```diff
@@ -0,0 +1 @@
+1.20.7
```
Comment (agnivade) [inline .go-version:1]: Can we move this file under the `server` directory?

**Review by agnivade — CHANGES_REQUESTED** [review]

File: .go-version (hunk 1 above)
Comment (hmhealey) [reply .go-version:1]: I think that'd depend on if there's any tooling that requires it to be at the root like with `.nvmrc`

Comment (agnivade) [reply .go-version:1]: AFAIK, there isn't anything like that. This file is equivalent to go.mod, and will eventually be replaced by that in Go 1.21. So it should be at the same level as that.

**Review by agnivade — APPROVED** [review]
/update-branch

**Pull request merged**